src/main/resources/db/changelog/version-1.0/19.07.2023.part1.create-product-table.sql

```diff
@@ -1,10 +1,12 @@
 CREATE TABLE IF NOT EXISTS product
 (
-    id          UUID,
-    name        VARCHAR(64) NOT NULL,
-    description TEXT,
-    price       DECIMAL     NOT NULL CHECK (price > 0),
-    quantity    INT         NOT NULL CHECK (quantity >= 0),
-    active      BOOLEAN     NOT NULL,
+    id             UUID,
+    name           VARCHAR(64) NOT NULL,
+    description    TEXT,
+    price          DECIMAL     NOT NULL CHECK (price > 0),
+    quantity       INT         NOT NULL CHECK (quantity >= 0),
+    active         BOOLEAN     NOT NULL,
+    average_rating DECIMAL     CHECK (average_rating >= 0 AND average_rating < 6) DEFAULT 0,
+    reviews_count  INT         CHECK (reviews_count >= 0) DEFAULT 0,
```

Comment:
Could you please clarify why is it necessary to store  `average_rating` and  `reviews_count` instead of calculating them on each request?

Reply:
I see know: it's either impossible or at least hacky to sort by a field that is not stored in the table